tests/test_core_behavior.py

```diff
@@ -0,0 +1,134 @@
+import sys
+from pathlib import Path
+from typing import cast
+
+import pytest
+
+sys.path.insert(0, str(Path(__file__).resolve().parent.parent / "src"))
```

Comment:
Again, this sort of thing shouldn't be necessary if you have the project installed properly, ie doing `uv pip install -e .` in project root directory.